poly/sharing_new.go

```
@@ -0,0 +1,161 @@
+
+// PriShare represents an individual private share v = p(i).
+type PriShare struct {
+	I int             // Index of the private share
+	V abstract.Scalar // Value of the private share
+}
+
+// PriPoly represents a secret sharing polynomial.
+type PriPoly struct {
+	g      abstract.Group    // Cryptographic group
+	coeffs []abstract.Scalar // Coefficients of the polynomial
+}
+
+// NewPriPoly creates a new secret sharing polynomial for the cryptographic
+// group g, the secret sharing threshold t, and the secret to be shared s.
+func NewPriPoly(g abstract.Group, t int, s abstract.Scalar) *PriPoly {
+	// XXX: old PriPoly.Pick()
+	return nil
+}
+
+// Threshold returns the secret sharing threshold.
+func (p *PriPoly) Threshold() int {
+	return len(p.coeffs)
+}
+
+// SharedSecret returns the shared secret p(0), i.e., the constant term of the polynomial.
+func (p *PriPoly) SharedSecret() abstract.Scalar {
+	// XXX: this is the old PriPoly.Secret()
+	return nil
+}
+
+// Eval computes the private share p(i).
+func (p *PriPoly) Eval(i int) *PriShare {
+	return nil
+}
+
+// Shares creates a list of n private shares p(1),...,p(n).
+func (p *PriPoly) Shares(n int) []PriShare {
+	// XXX: uses PriPoly.Eval()
+	// XXX: this is the old PriShare.Split()
+	return nil
+}
+
+// Add computes the component-wise sum of the polynomials p and q and returns it
+// as a new polynomial.
+func (p *PriPoly) Add(q *PriPoly) *PriPoly {
+	return nil
+}
+
+// Equal checks equality of two secret sharing polynomials p and q.
+func (p *PriPoly) Equal(q *PriPoly) bool {
+	return true
+}
+
+// Commit creates a public commitment polynomial for the given base point b.
+func (p *PriPoly) Commit(b abstract.Point) *PubPoly {
+	return nil
+}
+
+// XXX: Do we need that?
+func (p *PriPoly) String() String {
+	return nil
+}
+
+// xCoords creates an array of x-coordinates for Lagrange interpolation. In the
+// returned array, exactly t x-coordinates are non-nil.
+func priXCoords() []abstract.Scalar {
```

BTW, why does this `xCoords` panics instead of returning an error ? I don't see any reason why, so I would make it return an error instead.